### PR TITLE
feat: Better formatting for AggregateError

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/13831

This commit adds "aggregated" field to "deno_core::JsError" that stores
instances of "JsError" recursively to properly handle "AggregateError"
formatting. Appropriate logics was added to "PrettyJsError" and 
"console" API to format AggregateErrors.

### CODE REVIEW

**Comment by bartlomieju** [comment]
@nayeemrmn please take a look

**Comment by nayeemrmn** [comment]
I think it's noisy to show source lines for child errors (aggregated, cause). We should only show it for the main error, I implemented this in https://github.com/nayeemrmn/deno/commit/c18499bb047f9b2543f1e1b746096abbe3cf7be along with more fixes/cleanup.

**Comment by bartlomieju** [comment]
> I think it's noisy to show source lines for child errors (aggregated, cause). We should only show it for the main error, I implemented this in [nayeemrmn@c18499b](https://github.com/nayeemrmn/deno/commit/c18499bb047f9b2543f1e1b746096abbe3cf7be) along with more fixes/cleanup.

Thanks, these are great changes. I merged your commit into this PR.

File: core/error.rs [hunk 1]
```diff
@@ -104,6 +104,7 @@ pub struct JsError {
   pub frames: Vec<JsStackFrame>,
   pub source_line: Option<String>,
   pub source_line_frame_index: Option<usize>,
+  pub aggregated: Option<Vec<Box<JsError>>>,
```
Comment (littledivy) [inline core/error.rs:107]: Redundant `Option<Vec<Box<T>>>`? It can just be a `Vec<T>`.

Comment (bartlomieju) [reply core/error.rs:107]: It's not, this option allows distinction between AggregateError and regular Error.

Comment (littledivy) [reply core/error.rs:107]: Oh, I meant the redundant use of the `Box<T>`. compile-time size shouldn't be an issue since Vec is heap-allocated.

Comment (bartlomieju) [reply core/error.rs:107]: Good point. I'll remove these boxes

**Review by littledivy — APPROVED** [review]

**Pull request merged**